check_audio_framemd5.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+  SCRIPT=$(basename "${0}")
+ VERSION='2017-04-15'
+  AUTHOR='ffmprovisr'
+     RED='\033[1;31m'
+    BLUE='\033[1;34m'
+      NC='\033[0m'
+
+if [[ ${OSTYPE} = "cygwin" ]] || [ ! $(which diff) ]; then
+    echo -e "${RED}Error: 'diff' is not installed by default. Please install 'diffutils' from Cygwin.${NC}"
+    exit 1
+fi
+
+_output_prompt(){
+    cat <<EOF
+Usage: ${SCRIPT} [-h] | [ -i <av_file> -m <md5_file> ]
+EOF
+    exit 1
+}
+
+_output_help(){
+    cat <<EOF
+Syntax:
+  ${SCRIPT}
+    Prompts a short help message.
+  ${SCRIPT} -h
+    This help.
+  ${SCRIPT} -i <av_file> -m <md5_file>
+    Pass to the script the audio-visual file and the corresponding MD5
+    file to check.
+Dependency:
+  ffmpeg
+About:
+  Version: ${VERSION}
+  Website: https://github.com/amiaopensource/ffmprovisr/blob/gh-pages/check_framemd5.sh
+EOF
+    exit 0
+}
+
+unset input_file
+unset input_hash
+
+while getopts ":hi:m:" opt; do
+    case "${opt}" in
+        h) _output_help ;;
+        i) input_file=$OPTARG ;;
+        m) input_hash=$OPTARG ;;
+        :) echo -e "${RED}Error: option -${OPTARG} requires an argument${NC}" ; _output_prompt ;;
+        *) echo -e "${RED}Error: bad option -${OPTARG}${NC}" ; _output_prompt ;;
+    esac
+done
+
+[[ -z "${#}" || ! ${input_file} || ! ${input_hash} ]] && _output_prompt
+echo -e "${BLUE}Please wait...${NC}"
+unset md5_tmp
+if [[ $OSTYPE = "cygwin" ]]; then
+    md5_tmp=""${USERPROFILE}/$(basename ${input_hash}).tmp""
+else
+    md5_tmp="${HOME}/$(basename ${input_hash}).tmp"
+fi
+# Find audio frame size for hash calculation
+sample_rate=$(grep -v '^#' ${input_hash} | head -n 1 | tr -d ' ' | cut -d',' -f4)
```

Comment:
should put $input_hash in quotes throughout, else when you run `check_audio_framemd5.sh -m " ; find ~/Music -name *.mp3 -delete"` you might be sad.